### PR TITLE
[bsp]gd32105r-start, gd32205r-start, gd32305r-start

### PR DESCRIPTION
**Pull request by Ouxiaolong**
## 拉取/合并请求描述：(PR description)

更新GD32 BSP架构，增加gd32105r-start，gd32205r-start， gd32305r-start BSP
Update GD32 BSP architecture, add gd32105r-start, gd32205r-start, gd32305r-start BSP

已经在gd32105r-start, gd32205r-start, gd32305r-start BSP测试通过
Has been tested in gd32105r-start, gd32205r-start, gd32305r-start BSP

以下的内容不应该在提交PR时的message修改，修改下述message，PR会被直接关闭。请在提交PR后，浏览器查看PR并对以下检查项逐项check，没问题后逐条在页面上打钩。
The following content must not be changed in the submitted PR message. Otherwise, the PR will be closed immediately. After submitted PR, please use a web browser to visit PR, and check items one by one, and ticked them if no problem.

### 当前拉取/合并请求的状态 Intent for your PR

必须选择一项 Choose one (Mandatory):

- [ ] 本拉取/合并请求是一个草稿版本 This PR is for a code-review and is intended to get feedback
- [x] 本拉取/合并请求是一个成熟版本 This PR is mature, and ready to be integrated into the repo

### 代码质量 Code Quality：

我在这个拉取/合并请求中已经考虑了 As part of this pull request, I've considered the following:

- [ ] 已经仔细查看过代码改动的对比 Already check the difference between PR and old code
- [ ] 代码风格正确，包括缩进空格，命名及其他风格 Style guide is adhered to, including spacing, naming and other styles
- [ ] 没有垃圾代码，代码尽量精简，不包含`#if 0`代码，不包含已经被注释了的代码 All redundant code is removed and cleaned up
- [ ] 所有变更均有原因及合理的，并且不会影响到其他软件组件代码或BSP All modifications are justified and not affect other components or BSP
- [ ] 对难懂代码均提供对应的注释 I've commented appropriately where code is tricky
- [ ] 本拉取/合并请求代码是高质量的 Code in this PR is of high quality
- [ ] 本拉取/合并使用[formatting](https://github.com/mysterywolf/formatting)等源码格式化工具确保格式符合[RT-Thread代码规范](../documentation/coding_style_cn.md) This PR complies with [RT-Thread code specification](../documentation/coding_style_en.txt)

### CODE REVIEW

**Comment by liukangcc** [comment]
ci 加一下 https://github.com/RT-Thread/rt-thread/blob/master/.github/workflows/action.yml

File: bsp/gd32/libraries/gd32_drivers/drv_spi.c [hunk 1]
```diff
@@ -0,0 +1,356 @@
...
+static struct rt_spi_bus spi_bus2;
+#endif
+
+static const struct gd32_spi spi_bus_obj[] = {
+
+#ifdef BSP_USING_SPI0
+    {
+        SPI0,
+        "spi0",
+            RCU_SPI0,
```
Comment (Guozhanxin) [inline bsp/gd32/libraries/gd32_drivers/drv_spi.c]: 格式好像有些问题，最好用格式化工具刷一下

File: bsp/gd32/libraries/gd32_drivers/drv_spi.c [hunk 2]
```diff
@@ -0,0 +1,356 @@
...
+{
+    /* enable SPI clock */
+    rcu_periph_clock_enable(gd32_spi->spi_clk);
+    rcu_periph_clock_enable(gd32_spi->gpio_clk);
+
+#if defined SOC_SERIES_GD32F4xx
+        /*GPIO pin configuration*/
+    gpio_af_set(gd32_spi->spi_port, GPIO_AF_5,  gd32_spi->sck_pin | gd32_spi->mosi_pin | gd32_spi->miso_pin);
+
+      gpio_mode_set(gd32_spi->spi_port, GPIO_MODE_AF, GPIO_PUPD_NONE, gd32_spi->miso_pin| gd32_spi->miso_pin);
```
Comment (Guozhanxin) [inline bsp/gd32/libraries/gd32_drivers/drv_spi.c]: 格式问题挺多的，推荐用 astyle 这个工具

File: bsp/gd32/libraries/gd32_drivers/drv_spi.c (hunk 1 above)
Comment (Ouxiaolong) [reply bsp/gd32/libraries/gd32_drivers/drv_spi.c]: 刷了的，主要是一次提的太多

File: bsp/gd32/libraries/gd32_drivers/drv_spi.c (hunk 2 above)
Comment (Ouxiaolong) [reply bsp/gd32/libraries/gd32_drivers/drv_spi.c]: 全都改好了

**Pull request merged**